packages/cozy-device-helper/src/flagship.ts

```diff
@@ -15,7 +15,11 @@ export interface FlagshipMetadata {
   version?: string
 }
 
+const getGlobalWindow = (): Window =>
+  typeof window !== 'undefined' ? window : undefined
+
 export const getFlagshipMetadata = (): FlagshipMetadata =>
-  window.cozy?.flagship || {}
+  getGlobalWindow()?.window.cozy?.flagship || {}
```

Comment:
un `window?.cozy?.flagship || {}` n'aurait pas suffit ? 🤔

Reply:
Nop, car dans le cas de node, `window` ne vaut pas `undefined` mais n'existe carrément pas. Tu obtiens donc une erreur `window is not defined`. Le conditional chaining ne changera rien.

`typeof` sur une variable qui n'existe pas retournera `'undefined'` (note les guillemets autour)

Reply:
C'est la même raison pour laquelle le premier fix de cozy-client n'a pas fonctionné: https://github.com/cozy/cozy-client/pull/1176/files

Et qu'on a dû faire un nouveau fix ensuite: https://github.com/cozy/cozy-client/pull/1180/files#diff-9d26647e8d41ba1414842f5d3dca72250075f14efed52997e19584197a66e2a6

Reply:
Même retour que @acezard plus bas
```suggestion
  getGlobalWindow()?.cozy?.flagship || {}
```

Reply:
Woups effectivement ça doit être ça